File: fe-admin/src/app/core/model/response/permission-response.ts
import {RoleResponse} from "./role-response";

export interface PagePermission {
    pageCode: string;
    pageName: string;
    granted: { [actionCode: string]: boolean };
}

export interface PermissionResponse {
    role: RoleResponse;
    actions: string[];
    pages: PagePermission[];
}

File: be-api/src/main/java/cd/beapi/dto/response/PermissionResponse.java
package cd.beapi.dto.response;

import java.util.List;
import java.util.Map;

public record PermissionResponse(
        RoleResponse role,
        List<String> actions,
        List<PagePermission> pages
) {
    public record PagePermission(
            String pageCode,
            String pageName,
            Map<String, Boolean> granted
    ) {}
}
